core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java

```diff
@@ -1413,9 +1413,6 @@ private void saveTabStates() {
   public void onPause() {
     super.onPause();
     saveTabStates();
-    if (tts != null) {
```

Comment:
this will cause a memory leak, we probably want to stop it in onDestroy/View

Reply:
It is already being set to null in onDestroyView, then how leak may occur?

Reply:
I could be mistaken but we need to stop this task at some time. I believe it has a reference to the activity/fragment as it updates the view and if that is not relinquished then we could be in trouble.

Anyway it is easy enough to check, start reading aloud and press back until you exit the app

Reply:
* First thing, it doesn't update view while reading aloud(by updating view I mean, scrolling to the position where it is reading)
* As soon as I press back on reader with tts on, it stops. tts works in background when I press home button(os one)
* And there are so many leaks in the mainActivity and CoreReaderFrag. but non of 'em depends or emerges out of tts, they are mostly layouts(drawer and constraint) and ui-components(toolbar) being leaked.

Reply:
It absolutely updates the view because it controls the visibility of `pause/stop` buttons, updating the view is any change to the view hierarchy. I'll investigate later on but I am in the middle of another ticket, trying to remember the lifecycle of this thing is painful

Reply:
Well I'm not sure of what you're talking about but it doesn't scroll while reading when I run it on _Install Gentoo Wiki_, also when tts is running and the screen is left idle for some time, it(screen) goes off, then how can it(tts) update the view while reading if it can't keep the screen wake up while reading? Also, in my device it does not update view even when screen is awake.
So what should I do with it now?

Reply:
It hides and shows the pause/stop buttons, that is what I mean by "updating the view". 
>updating the view is any change to the view hierarchy

A non visible or partially visible view can be updated, wether you should or not is a different story. Partially visible == onPause, no longer visible (you pressed home lets say) == onStop. Then the app could be destroyed at any time if android feels like taking back memory, see [the lifecycle](https://developer.android.com/guide/components/activities/process-lifecycle)

Reply:
I couldn't find any leak there(emerging out of tts). Can you please check if I'm doing anything wrong. I followed the same procedure to produce leak what you mentioned earlier

Reply:
It all seems good. We may at some point want to move text to speech into a Service and surface the controls as a notification but this solves the issue

Reply:
Yeah, I thought of controlling tts via a notification like the other music applications provide control over music to user in background plays, like vlc, setting up notification is on easier side. But to boost the UX I thought of adding a floating widget like [musixmatch](https://play.google.com/store/apps/details?id=com.musixmatch.android.lyrify&hl=en_IN&gl=US), that has tts controls. What do you think is better to have?

Reply:
I'd go simple first, worry about the best possible UX in a v2